compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -303,6 +303,15 @@ pub enum StatementKind<'tcx> {
     /// and modify the place in an opaque way.
     Retag(RetagKind, Box<Place<'tcx>>),
 
+    /// This statement exists to preserve a trace of a scrutinee matched against a wildcard
+    /// binding.  This is especially useful for `let _ = PLACE;` bindings that desugar to a single
+    /// `PlaceMention(PLACE)`.
+    ///
+    /// When executed at runtime this is a nop.
```

Comment:
This needs to be updated.
```suggestion
    /// When executed at runtime, this computes the given place, but then discards
    /// it without doing a load. It is UB if the place is not pointing to live memory.
```

Reply:
AscribeUserType would probably need something similar, judging from the case you added in Miri?